baseplate/clients/thrift.py

```diff
@@ -69,6 +70,21 @@ class ThriftContextFactory(ContextFactory):
 
     """
 
+    PROM_PREFIX = "bp_thrift_pool"
+    PROM_LABELS = ["client_cls"]
+
+    promTotalConnections = Gauge(
+        f"{PROM_PREFIX}_size",
+        "Number of connections in this thrift pool",
+        PROM_LABELS,
+    )
+
+    promUsedConnections = Gauge(
+        f"{PROM_PREFIX}_in_use",
+        "Number of connections currently in use in this thrift pool",
+        PROM_LABELS,
+    )
```

Comment:
🔕 In Baseplate.go the equivalent (more or less, this is not really 100% mapped between go and py code) of these 2 gauges [are](https://github.com/reddit/baseplate.go/blob/ca159a8a3540e5154f5c6f7cde55c2f1ccdc7ae9/thriftbp/prometheus.go#L203-L215): `thriftbp_client_pool_allocated_clients` and `thriftbp_client_pool_active_connections` with `thrift_slug` label.

these are not core metrics defined in baseplate spec and do not have to be the same between go and py (as the implementation details differ), but just for your reference as it's likely still beneficial to make them as consistent as possible.

Reply:
Yes, it would be useful to determine if these metrics are equivalent enough to the go versions. Having them use the same prefix would be nice, as it would make comparing / alerting / dashboards easier.

Reply:
I did look at the redis implementation and they're unfortunately very different, but none of the others. Thrift is a likely candidate to be similar.